cluster/images/etcd/Makefile

```diff
@@ -50,16 +56,28 @@ build:
 ifeq ($(ARCH),amd64)
 
 	# Do not compile if we should make an image for amd64, use the official etcd binaries instead
-	curl -sSL --retry 5 https://github.com/coreos/etcd/releases/download/v$(TAG)/etcd-v$(TAG)-linux-amd64.tar.gz | tar -xz -C $(TEMP_DIR) --strip-components=1
+	for tag in $(TAGS); do \
+		tmp_dir=$(shell mktemp -d); \
+		curl -sSL --retry 5 https://github.com/coreos/etcd/releases/download/v$$tag/etcd-v$$tag-linux-amd64.tar.gz | tar -xz -C $$tmp_dir --strip-components=1; \
+		cp $$tmp_dir/etcd $$tmp_dir/etcdctl $(TEMP_DIR)/; \
+		cp $(TEMP_DIR)/etcd $(TEMP_DIR)/etcd-$$tag; \
+		cp $(TEMP_DIR)/etcdctl $(TEMP_DIR)/etcdctl-$$tag; \
+	done
 else
 	
 	# Download etcd in a golang container and cross-compile it statically
-	docker run -it -v $(TEMP_DIR):/etcdbin golang:$(GOLANG_VERSION) /bin/bash -c \
-		"git clone https://github.com/coreos/etcd /go/src/github.com/coreos/etcd \
-		&& cd /go/src/github.com/coreos/etcd \
-		&& git checkout v$(TAG) \
-		&& GOARM=$(GOARM) GOARCH=$(ARCH) ./build \
-		&& cp -f bin/$(ARCH)/etcd* bin/etcd* /etcdbin; echo 'done'"
+	for tag in $(TAGS); do \
+		tmp_dir=$(shell mktemp -d); \
+		docker run -it -v $$tmp_dir:/etcdbin golang:$(GOLANG_VERSION) /bin/bash -c \
+			"git clone https://github.com/coreos/etcd /go/src/github.com/coreos/etcd \
+			&& cd /go/src/github.com/coreos/etcd \
+			&& git checkout v$$tag \
+			&& GOARM=$(GOARM) GOARCH=$(ARCH) ./build \
+			&& cp -f bin/$(ARCH)/etcd* bin/etcd* /etcdbin; echo 'done'"; \
+		cp $$tmp_dir/etcd $$tmp_dir/etcdctl $(TEMP_DIR)/; \
+		cp $(TEMP_DIR)/etcd $(TEMP_DIR)/etcd-$$tag; \
+		cp $(TEMP_DIR)/etcdctl $(TEMP_DIR)/etcdctl-$$tag; \
+	done
```

Comment:
@lavalamp - sorry, I'm not sure I understand this question.

But if I understood your question correctly, the the script you are talking about is migrate_if_needed.sh that I'm working on https://github.com/kubernetes/kubernetes/pull/34009 (this will do the necessary migrations based on env variables that are set).

BTW - as part of Dockerfile/Makefile we don't start anything - all the logic what we should start is done in the manifest file. And I'm not changing this one - with this PR I'm just ensuring that all potentially necessary binaries are present in the image.

Or did I misunderstand your question?